worlds/lingo/player_logic.py

```diff
@@ -83,7 +83,8 @@ def set_door_item(self, room: str, door: str, item: str):
 
     def handle_non_grouped_door(self, room_name: str, door_data: Door, world: "LingoWorld"):
         if room_name in PROGRESSION_BY_ROOM and door_data.name in PROGRESSION_BY_ROOM[room_name]:
-            if room_name == "Orange Tower" and not world.options.progressive_orange_tower:
+            if (room_name == "Orange Tower" and not world.options.progressive_orange_tower)\
+                    or (room_name == "The Colorful" and not world.options.progressive_colorful):
```

Comment:
I'll let you get away with it but if you add like, a third and fourth thing, I'll ask you to find a way to format it differently